server/agent/callbacks.py

```diff
@@ -0,0 +1,42 @@
+from langchain_core.callbacks import BaseCallbackHandler
+from asgiref.sync import async_to_sync
+from channels.layers import get_channel_layer
+
+class WebSocketCallbackHandler(BaseCallbackHandler):
+    def __init__(self, group_name):
```

Comment:
This one could take a conversation as a second argument - then it can generate the group_name on it own